fastly/paginator.go

```diff
@@ -29,3 +37,178 @@ type PaginatorKVStoreEntries interface {
 	Keys() []string
 	Err() error
 }
+
+func NewPaginator[T ACLEntry | DictionaryItem | Service](client *Client, input *ListInput, path string) Paginator[T] {
+	return &ListPaginator[T]{
+		client: client,
+		input:  input,
+		path:   path,
+	}
+}
+
+type ListInput struct {
+	// Direction is the direction in which to sort results.
+	Direction string
+	// Page is the current page.
+	Page int
+	// PerPage is the number of records per page.
+	PerPage int
+	// Sort is the field on which to sort.
+	Sort string
+}
+
+// Paginator represents a generic paginator.
+type Paginator[T ACLEntry | DictionaryItem | Service] interface {
```

Comment:
Do you need this interface?  If nothing other than `ListPaginiator` implements it, it's probably not necessary.